sdk/azidentity/aad_identity_client.go

```diff
@@ -127,7 +164,32 @@ func (c *aadIdentityClient) createAccessToken(res *azcore.Response) (*azcore.Acc
 	// NOTE: look at go-autorest
 	accessToken.Token = value.Token
 	accessToken.ExpiresOn = time.Now().Add(time.Second * time.Duration(t)).UTC()
-	return accessToken, nil
+	return &tokenResponse{token: accessToken, refreshToken: value.RefreshToken}, nil
+}
+
+func (c *aadIdentityClient) createRefreshTokenRequest(tenantID, clientID, clientSecret, refreshToken string, scopes []string) (*azcore.Request, error) {
+	urlStr := c.options.AuthorityHost.String() + tenantID + tokenEndpoint
```

Comment:
Does this properly handle the case with a custom `AuthorityHost` that doesn't end with a trailing /?  You might want to use `ResolveReference()`, see https://stackoverflow.com/questions/34668012/combine-url-paths-with-path-join